integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/IndexedEmbeddedBaseIT.java

```diff
@@ -1617,10 +1822,682 @@ class EntityC {
+				.field( "indexedEntityString", String.class )
+				.objectField( "included", b2 -> b2
+						.field( "includedString", String.class )
+						.objectField( "subIncluded", b3 -> b3
+								.field( "subIncludedStringB", String.class )
+								.field( "subIncludedStringC", String.class )
+						)
+				)
+		);
+
+		SearchMapping mapping = setupHelper.start()
+				.withAnnotatedEntityTypes( Model.IndexedEntity.class )
+				.setup();
+
+		backendMock.verifyExpectationsMet();
+
+		Model model = new Model();
+
+		doTestEmbeddedRuntime(
+				mapping,
+				id -> model.new IndexedEntity(
+						id, "a",
+						model.new Included( "b", model.new SubIncluded(
+								"c", "d", "e" )
+						)
+				),
+				document -> document.field( "indexedEntityString", "a" )
+						.objectField( "included", b2 -> b2
+								.field( "includedString", "b" )
+								.objectField( "subIncluded", b3 -> b3
+										.field( "subIncludedStringB", "d" )
+										.field( "subIncludedStringC", "e" )
+								)
+						)
+		);
+	}
+
+	@Test
+	public void parentExcludeChildExcludeSameProperty() {
+		class Model {
+			@Indexed(index = INDEX_NAME)
+			class IndexedEntity {
+				@DocumentId
+				Integer id;
+				@KeywordField
+				String indexedEntityString;
+
+				@IndexedEmbedded(excludePaths = "subIncluded.subIncludedStringA")
+				Included included;
+
+				public IndexedEntity(Integer id, String indexedEntityString, Included included) {
+					this.id = id;
+					this.indexedEntityString = indexedEntityString;
+					this.included = included;
+				}
+			}
+
+			class Included {
+				@KeywordField
+				String includedString;
+
+				@IndexedEmbedded(excludePaths = "subIncludedStringA")
+				SubIncluded subIncluded;
+
+				public Included(String includedString, SubIncluded subIncluded) {
+					this.includedString = includedString;
+					this.subIncluded = subIncluded;
+				}
+			}
+
+			class SubIncluded {
+				@KeywordField
+				String subIncludedStringA;
+				@KeywordField
+				String subIncludedStringB;
+				@KeywordField
+				String subIncludedStringC;
+
+				public SubIncluded(String subIncludedStringA, String subIncludedStringB, String subIncludedStringC) {
+					this.subIncludedStringA = subIncludedStringA;
+					this.subIncludedStringB = subIncludedStringB;
+					this.subIncludedStringC = subIncludedStringC;
+				}
+			}
+		}
+
+		assertThatThrownBy( () -> setupHelper.start()
+				.withAnnotatedEntityTypes( Model.IndexedEntity.class )
+				.setup() )
+				.isInstanceOf( SearchException.class )
+				.satisfies( FailureReportUtils.hasFailureReport()
+						.typeContext( Model.IndexedEntity.class.getName() )
+						.failure(
+								"An @IndexedEmbedded defines excludePaths filters that do not match anything",
+								"Non-matching excludePaths filters: [subIncluded.subIncludedStringA].",
+								"Encountered field paths: [includedString, subIncluded, subIncluded.subIncludedStringA, subIncluded.subIncludedStringB, subIncluded.subIncludedStringC].",
```

Comment:
`subIncluded.subIncludedStringA` is non-matching but still encountered?

There's something wrong with the message at the very least, and possibly with the algorithm.
I'd expect paths that are excluded in children to simply not be reported to parents, leading to those paths not being marked as encountered in parents... ?